ambari-server/src/main/resources/common-services/RANGER/0.4.0/package/scripts/setup_ranger_xml.py

```diff
@@ -120,6 +120,13 @@ def setup_ranger_admin(upgrade_type=None):
     create_parents=True
   )
 
+  File(format('{ranger_conf}/ranger-admin-env.sh'),
+    content = format("export JAVA_HOME={java_home}"),
+    owner = params.unix_user,
+    group = params.unix_group,
+    mode = 0755
+  )
+
```

Comment:
Instead of doing it like this as a one-off, why not just have Ambari manage the -env.sh file you? That way, future changes can be made directly to the template and the file will just be written out as needed.

It would look something like
```
# The java implementation to use.
export JAVA_HOME={{java64_home}}
```

Reply:
Hi @jonathan-hurley, this fix is currently for Ambari-2.6.2 release only. A consolidated -env.sh file is being planned for next major Ambari release.

Reply:
I suppose since it's only 1 property doing it this way is fine for 2.6.2